### PR TITLE
Feature/589 edit skillset

### PR DESCRIPTION
**Pull request by MarcEgliP**

### CODE REVIEW

**Review by Robin481 — CHANGES_REQUESTED** [review]
Real nice functionality but I think we should cut back on the javascript used. See chat.

File: app/assets/stylesheets/styles.scss [hunk 1]
```diff
@@ -42,4 +47,12 @@ pzsh-topbar {
 
 .text-gray {
   color: $skills-text-gray;
-}
\ No newline at end of file
+}
+
+.table-row {
+  background-color: white !important;
```
[the remark this answers is not in the document]
Comment (MarcEgliP) [reply app/assets/stylesheets/styles.scss:53]: Not needed anymore

File: app/assets/stylesheets/styles.scss [hunk 2]
```diff
@@ -42,4 +47,12 @@ pzsh-topbar {
...
+}
+
+.table-row:hover {
+  background-color: $light !important;
```
[the remark this answers is not in the document]
Comment (MarcEgliP) [reply app/assets/stylesheets/styles.scss:57]: No

File: spec/controllers/skills_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+require 'rails_helper'
+
+describe SkillsController do
+  describe 'Update Skill' do
+    render_views
+
+    it 'should switch category' do
+      edited_skill = {**Skill.first.as_json.symbolize_keys,
+                      category_parent: Category.where(title: "System-Engineering")[0].id}
```
Comment (mtnstar) [inline spec/controllers/skills_controller_spec.rb:9]: acccess by fixtures ! skills(:system_engineering)

see fixture access helpers in rails doc

or just construct the hash here manually ... transforming given entry to params makes it less readable ...

File: spec/controllers/skills_controller_spec.rb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+      patch :update, params: {id: edited_skill[:id], skill: edited_skill, validate_only: true}
+      expect(response.body).to have_select("skill_category_id", with_options: [Category.where(title: "System-Engineering")[0].children[0].title])
```
Comment (mtnstar) [inline spec/controllers/skills_controller_spec.rb:11]: don't use where here again ... check `let` ... and also don't use where to find a single record ...
also find to use static text here ... instead of fetching the Category entry.

File: spec/domain/skills_filter_spec.rb [hunk 5]
```diff
@@ -18,10 +18,11 @@
       filter = SkillsFilter.new(Skill.all, 458108788, nil, nil)
       filteredSkills = filter.scope
 
-      expect(filteredSkills.count).to eq(3)
+      expect(filteredSkills.count).to eq(4)
```
Comment (mtnstar) [inline spec/domain/skills_filter_spec.rb:21]: camel_case_plleeeeeaaaaaaseeeee :camel: :camel: :camel: :camel: :camel: :camel: :camel:

File: spec/views/people/index_spec.rb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+
+xdescribe 'Check if people are listed', type: :system do
```
Comment (mtnstar) [inline spec/views/people/index_spec.rb:2]: is this test related to the current feature?

File: app/javascript/controllers/index.js [hunk 7]
```diff
@@ -4,8 +4,9 @@
 
 import { application } from "./application"
```
Comment (mtnstar) [inline app/javascript/controllers/index.js:6]: is this all still required?

File: package.json [hunk 8]
```diff
@@ -6,6 +6,7 @@
     "@hotwired/turbo-rails": "^7.3.0",
     "@popperjs/core": "^2.11.8",
     "@puzzleitc/puzzle-shell": "^4.1.2",
+    "@rails/request.js": "^0.0.9",
```
Comment (mtnstar) [inline package.json:9]: required ?

**Review by mtnstar — CHANGES_REQUESTED** [review]
nice, you're getting there! :partying_face: 

but still some things to make this bit more clean and pretty ...

go go go :rocket:

**Comment by mtnstar** [comment]
please also make sure we get an 404 when trying to access :edit/:update without turbo ...
respond_to in controller and/or format in routes

File: spec/domain/skills_filter_spec.rb (hunk 5 above)
Comment (MarcEgliP) [reply spec/domain/skills_filter_spec.rb:21]: snake_case :snake: :snake: :snake:

File: spec/views/people/index_spec.rb (hunk 6 above)
Comment (MarcEgliP) [reply spec/views/people/index_spec.rb:2]: No

File: app/javascript/controllers/index.js (hunk 7 above)
Comment (MarcEgliP) [reply app/javascript/controllers/index.js:6]: Yes

File: package.json (hunk 8 above)
Comment (MarcEgliP) [reply package.json:9]: No not anymore

**Review by Robin481 — APPROVED** [review]
